pkg/metallb/install.go

```diff
@@ -0,0 +1,124 @@
+package metallb
+
+import (
+	_ "embed"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+var (
+	//go:embed yaml/metallb-operator.yaml
+	metallbOperator string
+
+	//go:embed yaml/metallb.yaml
+	metallb string
+
+	metallbVersions = map[string]string{
+		"v4.9":  "metallb-operator.4.9.0-2023032815534",
+		"v4.10": "metallb-operator.4.10.0-202305032028",
+		"v4.11": "metallb-operator.4.11.0-202302271715",
+		"v4.12": "metallb-operator.4.12.0-202305102015",
+		"v4.13": "metallb-operator.v4.13.0-202304190216",
+	}
+)
+
+func InstallIfNotExist(t test.TestHelper, kubeConfigs ...string) {
+	if len(kubeConfigs) == 0 {
+		installWithOC(t, *oc.DefaultOC)
+		return
+	}
+
+	for _, config := range kubeConfigs {
+		oc := oc.WithKubeconfig(config)
+		if oc == nil {
+			t.Errorf("failed to create oc config from kubeconfig file: %s", config)
+		}
+		installWithOC(t, *oc)
+	}
+}
+
+func installWithOC(t test.TestHelper, oc oc.OC) {
+	installOperator(t, oc)
+	deployMetalLB(t, oc)
+	createAddressPool(t, oc)
+}
+
+func installOperator(t test.TestHelper, oc oc.OC) {
+	t.Log("Check if MetalLB controller already exists")
+	if oc.ResourceExists(t, ns.MetalLB, "deployments", "metallb-operator-controller-manager") {
+		t.Log("MetalLB operator already exists - skip installation of the operator")
+		return
+	}
+
+	t.Log("Install MetalLB operator")
+	//nolint:typecheck
+	ocpVersion := version.ParseOCPVersion(oc.GetOCPVersion(t))
+	metallbVersion := metallbVersions[ocpVersion.String()]
+	oc.ApplyTemplateString(t, ns.MetalLB, metallbOperator, map[string]string{"Version": metallbVersion})
+	retry.UntilSuccess(t, func(t test.TestHelper) {
+		if !oc.ResourceExists(t, ns.MetalLB, "deployments", "metallb-operator-controller-manager") {
+			t.Log("metallb-operator-controller-manager not found - waiting until exists")
```

Comment:
To make this retry work is need to use instead of t.Log a t.Fatal because the retry needs a failure to make the retry, right now will log this only and continue with the rest of the steps

Reply:
Oh, right. Thanks!

Reply:
Done